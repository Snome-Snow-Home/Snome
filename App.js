import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import CreateUser from './components/CreateUser';

export default function App() {
  return (
    <View style={styles.container}>
      <StatusBar style="auto" />
      <CreateUser/>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
