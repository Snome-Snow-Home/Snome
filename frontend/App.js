import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
// import { NavigationContainer } from '@react-navigation/native';
import HomeScreen from './src/screens/TownsScreen';
// import CreateUser from './src/components/CreateUser';
import { ThemeProvider } from 'styled-components/native';

import { theme } from './src/theme';

import { Navigation } from './src/navigation';

export default function App() {

  // sessions handler
  // renders <CreateUser/> if no session is found
  // *remember to uncomment import to use the component*

  return (
    <>
      <ThemeProvider theme={theme}>
        <Navigation />
      </ThemeProvider>
      <StatusBar style="auto" />
    </>
  );
}

// const styles = StyleSheet.create({
//   container: {
//     flex: 1,
//     backgroundColor: '#fff',
//     alignItems: 'center',
//     justifyContent: 'center',
//   },
// });
