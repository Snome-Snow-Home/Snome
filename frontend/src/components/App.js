import { StatusBar } from 'expo-status-bar';
import React, { useEffect, useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import CreateUser from './components/CreateUser';

export default function App() {
  return (
    <View style={styles.container}>
      <CreateUser />
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    // flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
