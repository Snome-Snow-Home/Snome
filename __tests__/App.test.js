import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('react-native', () => {
  return {
    StyleSheet: {
      create: () => ({}),
    },
  };
});

describe('<App />', () => {
  // it('has 2 children', () => {
  //   const tree = renderer.create(<App />).toJSON();
  //   expect(tree.children.length).toBe(2);
  // });



  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

