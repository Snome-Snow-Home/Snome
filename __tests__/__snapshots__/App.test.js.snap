// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <div
    style={
      Object {
        "maxWidth": "400px",
        "width": "95%",
      }
    }
  >
    <img
      src={1}
      style={
        Object {
          "width": "150px",
        }
      }
    />
    <h2>
      New User? Sign up here
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-gKclnd ejFowS"
      >
        <label
          className="sc-eCImPb jcRukK"
          htmlFor="name"
        >
          Name: 
        </label>
        <div
          className="sc-jRQBWg gyalWg"
        >
          *Required
        </div>
      </div>
      <TextInput
        allowFontScaling={true}
        autoCorrect="false"
        id="name"
        onChange={[Function]}
        placeholder="Name"
        rejectResponderTermination={true}
        required={true}
        style={Object {}}
        type="text"
        underlineColorAndroid="transparent"
        value=""
      />
      <div>
        <br />
      </div>
      <div
        className="sc-gKclnd ejFowS"
      >
        <label
          className="sc-eCImPb jcRukK"
          htmlFor="email"
        >
          Email: 
        </label>
        <div
          className="sc-jRQBWg gyalWg"
        >
          *Required
        </div>
      </div>
      <TextInput
        allowFontScaling={true}
        id="email"
        onChange={[Function]}
        placeholder="Email"
        rejectResponderTermination={true}
        required={true}
        style={Object {}}
        type="text"
        underlineColorAndroid="transparent"
        value=""
      />
      <div>
        <br />
      </div>
      <div
        className="sc-gKclnd ejFowS"
      >
        <label
          className="sc-eCImPb jcRukK"
          htmlFor="address"
        >
          Address: 
        </label>
        <div
          className="sc-jRQBWg gyalWg"
        >
          *Required
        </div>
      </div>
      <TextInput
        allowFontScaling={true}
        id="address"
        onChange={[Function]}
        placeholder="Address"
        rejectResponderTermination={true}
        required={true}
        style={Object {}}
        type="text"
        underlineColorAndroid="transparent"
        value=""
      />
      <div>
        <br />
      </div>
      <div
        className="sc-gKclnd ejFowS"
      >
        <label
          className="sc-eCImPb jcRukK"
          htmlFor="city"
        >
          City: 
        </label>
        <div
          className="sc-jRQBWg gyalWg"
        >
          *Required
        </div>
      </div>
      <TextInput
        allowFontScaling={true}
        id="city"
        onChange={[Function]}
        placeholder="City"
        rejectResponderTermination={true}
        required={true}
        style={Object {}}
        type="text"
        underlineColorAndroid="transparent"
        value=""
      />
      <div>
        <br />
      </div>
      <div
        className="sc-gKclnd ejFowS"
      >
        <label
          className="sc-eCImPb jcRukK"
          htmlFor="state"
        >
          State: 
        </label>
        <div
          className="sc-jRQBWg gyalWg"
        >
          *Required
        </div>
      </div>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        id="state"
        maxLength="2"
        onChange={[Function]}
        placeholder="State"
        rejectResponderTermination={true}
        required={true}
        style={Object {}}
        type="text"
        underlineColorAndroid="transparent"
        value=""
      />
      <div>
        <br />
      </div>
      <div
        className="sc-gKclnd ejFowS"
      >
        <label
          className="sc-eCImPb jcRukK"
          htmlFor="zipCode"
        >
          Zip Code: 
        </label>
        <div
          className="sc-jRQBWg gyalWg"
        >
          *Required
        </div>
      </div>
      <TextInput
        allowFontScaling={true}
        id="zipCode"
        onChange={[Function]}
        placeholder="Zip Code"
        rejectResponderTermination={true}
        required={true}
        style={Object {}}
        type="text"
        underlineColorAndroid="transparent"
        value=""
      />
      <div>
        <br />
      </div>
      <div
        className="sc-gKclnd ejFowS"
      >
        <label
          className="sc-eCImPb jcRukK"
          htmlFor="password"
        >
          Password: 
        </label>
        <div
          className="sc-jRQBWg gyalWg"
        >
          *Required
        </div>
      </div>
      <TextInput
        allowFontScaling={true}
        autoComplete="new-password"
        id="password"
        onChange={[Function]}
        placeholder="Need a number, a letter and a special character"
        rejectResponderTermination={true}
        required={true}
        secureTextEntry="true"
        style={Object {}}
        type="password"
        underlineColorAndroid="transparent"
        value=""
      />
      <div>
        <br />
      </div>
      <div
        className="sc-gKclnd ejFowS"
      >
        <label
          className="sc-eCImPb jcRukK"
          htmlFor="confirmPassword"
        >
          Confirm Password: 
        </label>
        <div
          className="sc-jRQBWg gyalWg"
        >
          *Required
        </div>
      </div>
      <TextInput
        allowFontScaling={true}
        autoComplete="new-password"
        id="confirmPassword"
        onChange={[Function]}
        placeholder="Confirm Password"
        rejectResponderTermination={true}
        required={true}
        secureTextEntry="true"
        style={Object {}}
        type="password"
        underlineColorAndroid="transparent"
        value=""
      />
      <div>
        <br />
      </div>
      <button
        title="Submit"
        type="submit"
      >
        Submit
      </button>
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </form>
  </div>
</View>
`;
